internals-js/src/definitions.ts

```diff
@@ -2818,11 +2829,15 @@ export class VariableDefinition extends DirectiveTargetElement<VariableDefinitio
 
   toVariableDefinitionNode(): VariableDefinitionNode {
     return {
-      kind: 'VariableDefinition',
+      kind: Kind.VARIABLE_DEFINITION,
       variable: this.variable.toVariableNode(),
       type: typeToAST(this.type),
-      defaultValue: valueToAST(this.defaultValue, this.type),
-      directives: this.appliedDirectivesToDirectiveNodes()
+      // TODO: This casting is necessary because valueToAST can return a VariableNode
+      //       which is not a member of ConstNodeValue. I'm not sure if the right thing to do here
+      //       is to have an intermediate function that transforms ValueNodes into ConstValueNodes
+      //       and replaces all VariableNodes with StringValueNodes
+      defaultValue: valueToAST(this.defaultValue, this.type) as ConstValueNode,
```

Comment:
I'm not familiar with this code (and the requirements of this function), but an intermediate function which guarantees the output and its type is as expected seems correct. Is there a runtime concern here or do we happen to know more about this value than the compiler can (and can safely cast)?

Reply:
I guess the right thing to do is to make an alternative version that constructs a const version of the AST? Everything is exactly the same except for the type information. For example:

```typescript
export interface ListValueNode {
  readonly kind: Kind.LIST;
  readonly loc?: Location;
  readonly values: ReadonlyArray<ValueNode>;
}

export interface ConstListValueNode {
  readonly kind: Kind.LIST;
  readonly loc?: Location;
  readonly values: ReadonlyArray<ConstValueNode>;
}
```

So I don't really have a strong feel for whether or not it's necessary. I'm going to dig into how hard it would be to modify the existing function without duplicating code.

Reply:
Ok. These `Const` AST nodes are new to me, I'm comfortable deferring to your best judgment here. Both paths seems reasonable.

Reply:
After diving back in, there isn't a `VariableNode` in `ConstValueNode`, so I think we have to do the translation. Take a look at my latest changeset and let me know what you think.